src/main.rs

```diff
@@ -1,26 +1,57 @@
+#[derive(Debug, Clone)]
+struct Todo {
+    message: String,
+}
+
+use std::io::{Stdin, Stdout, Write};
+
+struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
+}
+
 fn main() {
     loop {
+        let mut pergunta = Terminal::new();
+        let todo = pergunta.ask_for_new_todo();
+
+        pergunta.show_todo(&todo);
+    }
+}
+
+impl Terminal {
+    fn new() -> Self {
+        Terminal {
+            stdin: std::io::stdin(),
+            stdout: std::io::stdout(),
+        }
+    }
+
+    fn ask_for_new_todo(&mut self) -> Todo {
         println!("\nQuer adicionar um novo TODO 📝?");
 
         println!("digite (sim) para confirmar 👍  ou (nao) para negar 👎");
 
-        let resposta = input();
+        let mut resposta = String::new();
+
+        self.stdin.read_line(&mut resposta).expect("sim");
 
-        if resposta == "sim" {
+        if resposta.trim() == "sim" {
             println!("\nQual TODO 📝 deseja criar?");
 
-            let todo = input();
+            let mut novo_todo = String::new();
 
-            println!("\nvocê criou o TODO\n\n 🔹 {}", todo);
+            self.stdin.read_line(&mut novo_todo).unwrap().to_string();
```

Comment:
Não entendi muito bem esse `.to_string()` aqui.

Reply:
inicialmente estava sem, aí acusou um erro, quando eu coloquei esse to string saiu, eu fiquei sem entender, mas deixei. Vou olhar agora e refazer :)